harper-core/src/word_metadata.rs

```diff
@@ -28,7 +31,28 @@ pub struct WordMetadata {
     #[serde(default = "default_false")]
     pub common: bool,
     #[serde(default = "default_none")]
-    pub derived_from: Option<WordId>,
+    pub derived_from: Option<HashSet<WordId>>,
```

Comment:
Is it really a good idea that we now need an allocation for almost every word token? This seems like it'll impact perf quite a bit.

Isn't this hash set pretty much constant for each word? If so, couldn't we replace it with an `Arc<HashSet<WordId>>`? (Or even more efficient: as an index + len pair into some global list.)

Reply:
> Is it really a good idea that we now need an allocation for almost every word token? This seems like it'll impact perf quite a bit.
> 
> Isn't this hash set pretty much constant for each word? If so, couldn't we replace it with an `Arc<HashSet<WordId>>`? (Or even more efficient: as an index + len pair into some global list.)

Oh I wonder if that's my patch I mentioned gone in? There's been a lot to try to keep up with today. The set is deficient in other ways too. It's only added to for derived forms so when a derived form of one word is the same as a dictionary word only the derived goes in.

I found the code a little messy by Harper's standards plus I didn't know what the intended final uses cases were. @elijah-potter ?